src/plugins/intel_gpu/src/kernel_selector/cl_kernels/scatter_elements_update_ref.cl

```diff
@@ -108,98 +171,154 @@ KERNEL(scatter_elements_update_ref)(OPTIONAL_SHAPE_INFO_ARG
-    #elif   AXIS_VALUE == 3
-        if (index < 0) { index += INPUT0_SIZE_Z; }
-        const uint x = idx_x; const uint y = idx_y; const uint z = index; const uint w = idx_w; const uint f = idx_f; const uint b = idx_b;
-    #elif   AXIS_VALUE == 4
-        if (index < 0) { index += INPUT0_SIZE_Y; }
-        const uint x = idx_x; const uint y = index; const uint z = idx_z; const uint w = idx_w; const uint f = idx_f; const uint b = idx_b;
-    #elif   AXIS_VALUE == 5
-        if (index < 0) { index += INPUT0_SIZE_X; }
-        const uint x = index; const uint y = idx_y; const uint z = idx_z; const uint w = idx_w; const uint f = idx_f; const uint b = idx_b;
-    #endif  // AXIS_VALUE
-    #endif
+    if (index < 0) { index += SIZE; }
+    ASSIGN_INDEX(index);
     const uint output_idx = GET_OUTPUT_INDEX(ORDER);
+    output[output_idx] = REDUCTION_NEUTRAL_VALUE;
+            }
+        }
+    }
+    #endif
+    #endif
+    #ifdef REDUCE_MODE  // A loop is needed for a single worker item in reduce
+    for (uint gz = 0; gz < tgz; gz++) {
+        for (uint gy = 0; gy < tgy; gy++) {
+            for (uint gx = 0; gx < tgx; gx++) {
+                uint ORDER;
+                #if OUTPUT_DIMS == 4
+                    x = gx;
+                    y = gy;
+                #elif OUTPUT_DIMS == 5
+                    x = gx % INPUT2_SIZE_X;
+                    y = gx / INPUT2_SIZE_X;
+                    z = gy;
+                #elif OUTPUT_DIMS == 6
+                    x = gx % INPUT2_SIZE_X;
+                    y = gx / INPUT2_SIZE_X;
+                    z = gy % INPUT2_SIZE_Z;
+                    w = gy / INPUT2_SIZE_Z;
+                #endif
+                f = gz % INPUT2_FEATURE_NUM;
+                b = gz / INPUT2_FEATURE_NUM;
+    #else // Multiple worker items do not need a loop when REDUCE_MODE==NONE.
+        uint ORDER;
+        #if OUTPUT_DIMS == 4
+            x = dim0;
+            y = dim1;
+            f = dim2 % INPUT2_FEATURE_NUM;
+            b = dim2 / INPUT2_FEATURE_NUM;
+        #elif OUTPUT_DIMS == 5
+            x = dim0 % INPUT2_SIZE_X;
+            y = dim0 / INPUT2_SIZE_X;
+            z = dim1;
+            f = dim2 % INPUT2_FEATURE_NUM;
+            b = dim2 / INPUT2_FEATURE_NUM;
+        #elif OUTPUT_DIMS == 6
+            x = dim0 % INPUT2_SIZE_X;
+            y = dim0 / INPUT2_SIZE_X;
+            z = dim1 % INPUT2_SIZE_Z;
+            w = dim1 / INPUT2_SIZE_Z;
+            f = dim2 % INPUT2_FEATURE_NUM;
+            b = dim2 / INPUT2_FEATURE_NUM;
+        #endif
+    #endif
 
-    const uint updates_idx = GET_UPDATES_INDEX(IDX_ORDER);
+    const uint indices_idx = GET_INDICES_INDEX(ORDER);
+    const uint updates_idx = GET_UPDATES_INDEX(ORDER);
     INPUT2_TYPE val = updates[(int)updates_idx];
+    INPUT1_TYPE index = indices[(int)indices_idx];
+    if (index < 0) {index += SIZE;}
```

Comment:
Does this work for >= 2d cases? 
Also there are only 1d tests, please add more dimensions cases.